src/tree/tenants/TenantResourceTreeDataProvider.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { TenantIdDescription } from '@azure/arm-resources-subscriptions';
+import { AzureSubscriptionProvider } from '@microsoft/vscode-azext-azureauth';
+import { nonNullProp, nonNullValueAndProp } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { TenantResourceProviderManager } from '../../api/ResourceProviderManagers';
+import { ext } from '../../extensionVariables';
+import { BranchDataItemCache } from '../BranchDataItemCache';
+import { GenericItem } from '../GenericItem';
+import { OnGetChildrenBase, getAzureSubscriptionProvider } from '../OnGetChildrenBase';
+import { ResourceGroupsItem } from '../ResourceGroupsItem';
+import { ResourceTreeDataProviderBase } from "../ResourceTreeDataProviderBase";
+import { TreeItemStateStore } from '../TreeItemState';
+import { TenantResourceBranchDataProviderManager } from "./TenantResourceBranchDataProviderManager";
+import { TenantTreeItem } from './TenantTreeItem';
+
+export class TenantResourceTreeDataProvider extends ResourceTreeDataProviderBase {
+    public subscriptionProvider: AzureSubscriptionProvider | undefined;
+    public statusSubscription: vscode.Disposable | undefined;
+
+    constructor(
+        protected readonly branchDataProviderManager: TenantResourceBranchDataProviderManager,
+        onRefresh: vscode.Event<void | ResourceGroupsItem | ResourceGroupsItem[] | null | undefined>,
+        protected readonly resourceProviderManager: TenantResourceProviderManager,
+        state: TreeItemStateStore,
+        branchItemCache: BranchDataItemCache,
+        callOnDispose?: () => void) {
+        super(
+            branchItemCache,
+            branchDataProviderManager.onDidChangeTreeData,
+            resourceProviderManager.onDidChangeResourceChange,
+            onRefresh,
+            state,
+            () => {
+                this.statusSubscription?.dispose();
+                callOnDispose?.();
+            });
+    }
+
+    async onGetChildren(element?: ResourceGroupsItem | undefined): Promise<ResourceGroupsItem[] | null | undefined> {
+        if (element) {
+            return await element.getChildren();
+        } else {
+            const subscriptionProvider = await getAzureSubscriptionProvider(this);
+            const children: ResourceGroupsItem[] = await OnGetChildrenBase(subscriptionProvider);
+
+            if (children.length === 0) {
+                const session = await vscode.authentication.getSession('microsoft', ['https://management.azure.com/.default']); // Will not work for sovereign clouds
```

Comment:
Is this a TODO?

Reply:
Somewhat the VSCode team is making some changes to how getSession works so it's likely this line will look different. We will have to see once they release it on insiders hopefully in the next couple of weeks 😊

Reply:
See https://github.com/microsoft/vscode/issues/213947. Tyler is working on updating/adding APIs to make interacting with multiple accounts easier.